packages/@react-aria/gridlist/src/useGridListSection.ts

```diff
@@ -37,20 +42,62 @@ export interface GridListSectionAria {
  */
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export function useGridListSection<T>(props: AriaGridListSectionProps, state: ListState<T>, ref: RefObject<HTMLElement | null>): GridListSectionAria {
-  let {'aria-label': ariaLabel} = props;
+  let {'aria-label': ariaLabel, node, isVirtualized} = props;
   let headingId = useSlotId();
   let labelProps = useLabels({
     'aria-label': ariaLabel,
     'aria-labelledby': headingId
   });
+  let rowIndex;
+
+  let sumOfNodes = (node: RSNode<unknown>): number => {
+    if (node.prevKey === null) {
+      let lastChildKey = (node as CollectionNode<T>).lastChildKey;
```

Comment:
if we want to use `lastChildKey`, the node types needs to defined as a `CollectionNode<T>`. however, `createBranchComponent` expects the `item` to be type `Node<T>` which doesn't have lastChildKey defined and i don't think i should change the type declarations in `createBranchComponent`. as a result, i end up having to cast the type here as `CollectionNode`. it should be fine tho since the BaseCollection uses CollectionNodes anyway